mod/consensus/pkg/cometbft/service/abci.go

```diff
@@ -204,13 +204,23 @@ func (s *Service[LoggerT]) PrepareProposal(
 		),
 	)
 
+	var slotData *types.SlotData[
+		*ctypes.AttestationData,
+		*ctypes.SlashingInfo,
+	]
+	slotData = slotData.New(
+		math.Slot(req.GetHeight()),
+		nil,
+		nil,
+
+		// we do not add h.minPayloadDelay here since req.GetTime()
+		// is guaranteed to be strictly larger than
+		// prevBlock.GetTime() + h.minPayloadDelay
```

Comment:
can you quickly remind me why this is the case? Is it because we factor it in to the nextPayloadTime [here](https://github.com/berachain/beacon-kit/pull/2095/files#diff-877718cea28b2a3ddf39d2cf7b9d5ad968d1a209fc23017de98e6cb0870b6b2fR227)?

Reply:
Yes. The req.Time here is the time of the block to be built.
CometBFT consensus sets this time to be larger than the nextPayloadTime, as we set in Process/FinalizeBlock calls before (which were referring to the previous block)